ql/src/java/org/apache/hadoop/hive/ql/optimizer/topnkey/TopNKeyPushdownProcessor.java

```diff
@@ -244,18 +223,41 @@ private void pushdownThroughLeftOuterJoin(TopNKeyOperator topNKey) throws Semant
             reduceSinkDesc.getColumnExprMap(),
             reduceSinkDesc.getOrder(),
             reduceSinkDesc.getNullOrder());
+
+    pushDownThrough(commonKeyPrefix, topNKey, join, reduceSinkOperator);
+  }
+
+  private <T extends AbstractOperatorDesc> void pushDownThrough(
+          CommonKeyPrefix commonKeyPrefix, TopNKeyOperator topNKey, Operator<T> operator)
+          throws SemanticException {
+
+    pushDownThrough(commonKeyPrefix, topNKey, operator, operator);
+  }
+
+  private <TDesc extends AbstractOperatorDesc, TParentDesc extends AbstractOperatorDesc> void pushDownThrough(
+          CommonKeyPrefix commonKeyPrefix, TopNKeyOperator topNKey,
+          Operator<TDesc> operator, Operator<TParentDesc> parentOperator)
```

Comment:
Are there any advantages in using type parameters in this method (i.e., `TDesc`, `TParentDesc`) instead of wildcards? 

It seems that the conf of the operator is not used anywhere in this method so we could possibly simplify the signature to:
`private void pushDownThrough(CommonKeyPrefix commonKeyPrefix, TopNKeyOperator topNKey, Operator<?> operator, Operator<?> parentOperator)`